### PR TITLE
Inital setup of docusaurus

### PR DESCRIPTION
**Pull request by sveinpg**
#3161 

Added inital setup of docusaurus. 

Missing files in docs:  
- `README.md`
- `PATRONS.md`
- `CHANGELOG.md`

Docusaurus requires docs to be placed in `/docs`. Should we move these files?

Also, since docusaurus currently does not support nested sidebars, I moved `Structuring Reducers` one level up in the sidebar.

### CODE REVIEW

**Comment by markerikson** [comment]
Hmm.  Well, we can't move the _actual_ README, because that's gotta stay at the root of the repo. PATRONS could be moved, and tbh CHANGELOG is kinda pointless.  

Tell you what.  Let's _not_ include PATRONS or CHANGELOG in the actual published docs.  Leave the files there for historical purposes, but no point in publishing them.

For the README... do we even want the entire README as a docs page itself?  I'm not sure we do.  Perhaps we should have a cut-down version of it in the docs.

File: website/siteConfig.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    ],
+    myOtherFont: [
+      "-apple-system",
+      "system-ui"
+    ]
+  },
+  */
+
+  // This copyright info is used in /core/Footer.js and blog RSS/Atom feeds.
+  copyright: `Copyright © ${new Date().getFullYear()} Your Name or Your Company Name`,
```
Comment (timdorr) [inline website/siteConfig.js]: Need to fill this in.

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,9 @@ dist
 lib
 es
 coverage
+
+website/translated_docs
+website/build/
+website/yarn.lock
```
Comment (timdorr) [inline .gitignore]: Please don't use Yarn. We use npm here. Also, don't ignore the lock file generated.

Comment (sveinpg) [reply .gitignore]: Noted

File: website/siteConfig.js (hunk 1 above)
Comment (sveinpg) [reply website/siteConfig.js]: What do you want it to say? "Redux JS"?

Comment (timdorr) [reply website/siteConfig.js]: As per the license (which covers the existing docs), `Copyright (c) 2015-present Dan Abramov`.

Comment (timdorr) [reply website/siteConfig.js]: Actually, maybe we should get a little more specific. Dan doesn't own copyright for things he didn't contribute. `Copyright (c) 2015-present Dan Abramov and the Redux documentation authors.`

Comment (sveinpg) [reply website/siteConfig.js]: That's nice. Fixed now.

**Comment by sveinpg** [comment]
Should we create an issue to cover what needs to be done?

**Comment by markerikson** [comment]
Looks like Docusaurus just merged in the nested categories feature in https://github.com/facebook/Docusaurus/pull/892, and v1.4.1 should be out with that in the next few days.

File: docs/introduction/CoreConcepts.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
+---
+id: core-consepts
+title: Core consepts
+sidebar_label: Core consepts
```
Comment (timdorr) [inline docs/introduction/CoreConcepts.md]: Spelling: "concepts"

File: docs/recipes/reducers/PrerequisiteConcepts.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+---
+id: prerequisite-consepts
+title: Prerequisite Consepts
+sidebar_label: Prerequisite Consepts
```
Comment (timdorr) [inline docs/recipes/reducers/PrerequisiteConcepts.md]: Spelling: "concepts"

File: website/core/Footer.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                alt={this.props.config.title}
+                width="66"
+                height="58"
+              />
+            )}
+          </a>
+          <div>
+            <h5>Docs</h5>
+            <a href={this.docUrl("introduction/core-consepts")}>
+              Core consepts
```
Comment (timdorr) [inline website/core/Footer.js]: Spelling: "concepts"

File: website/pages/en/index.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+class HomeSplash extends React.Component {
+  render() {
+    const language = this.props.language || "";
+    return (
+      <SplashContainer>
+        <div className="inner">
+          <ProjectTitle />
+          <PromoSection>
+            <Button href={docUrl("introduction/core-consepts", language)}>
+              Core consepts
```
Comment (timdorr) [inline website/pages/en/index.js]: Spelling: "concepts"

File: website/sidebars.json [hunk 7]
```diff
@@ -0,0 +1,81 @@
+{
+  "docs": {
+    "Introduction": [
+      "introduction/motivation",
+      "introduction/core-consepts",
```
Comment (timdorr) [inline website/sidebars.json]: Spelling: "concepts"

File: website/sidebars.json [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      "recipes/reducing-boilerplate",
+      "recipes/server-rendering",
+      "recipes/writing-tests",
+      "recipes/computing-derived-data",
+      "recipes/implementing-undo-history",
+      "recipes/using-immutablejs-with-redux"
+    ],
+    "Structuring Reducers": [
+      "recipes/structuring-reducers",
+      "recipes/reducers/prerequisite-consepts",
```
Comment (timdorr) [inline website/sidebars.json]: Spelling: "concepts"

File: website/siteConfig.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  // Used for publishing and more
+  projectName: "redux",
+  organizationName: "reduxjs",
+  // For top-level user or org sites, the organization is still the same.
+  // e.g., for the https://JoelMarcey.github.io site, it would be set like...
+  //   organizationName: 'JoelMarcey'
+
+  // For no header links in the top nav bar -> headerLinks: [],
+  headerLinks: [
+    { doc: "introduction/core-consepts", label: "Docs" },
```
Comment (timdorr) [inline website/siteConfig.js]: Spelling: "concepts"

**Comment by selbekk** [comment]
What a great god damn effort @sveinpg :clap: :clap: :clap:

**Comment by timdorr** [comment]
The only thing I don't like about this site so far: The sidebar position resets after navigating to a new page. If I'm way down at the API reference and click over from createStore to the Store, the sidebar is back at the top. 

Also, for the "README", I'd like to see that be a "Getting Started" page that can replace the "Core Concepts" button on the front page. And that should contain just the bare minimum to make a store and dispatch actions. Simple and straightforward for those that don't want to read through many pages of docs 😄

**Comment by endiliey** [comment]
@timdorr

There is a workaround for the sidebar scroll position

https://github.com/facebook/Docusaurus/issues/823#issuecomment-421152269

**Comment by yangshun** [comment]
@sveinpg Hi there, Docusaurus maintainer here. We managed to get the subcategories feature shipped in https://github.com/facebook/Docusaurus/pull/1026 and you should be able to keep the subcategories for "Structuring Reducers". Could you point your Docusaurus version to trunk for the moment? If there are things that Redux needs but Docusaurus doesn't support we could do it over the weekend. Else if all is good, I'll release Docusaurus 1.5.0 over the weekend which will include the subcategories feature.

Let me know what you think! We're here to support Redux's migration to Docusaurus 😄

**Comment by sveinpg** [comment]
@yangshun How do I point my docusaurus version to trunk?

File: website/package.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  "scripts": {
+    "start": "docusaurus-start",
+    "build": "docusaurus-build",
+    "publish-gh-pages": "docusaurus-publish",
+    "write-translations": "docusaurus-write-translations",
+    "version": "docusaurus-version",
+    "rename-version": "docusaurus-rename-version"
+  },
+  "devDependencies": {
+    "docusaurus": "^1.4.0"
```
Comment (yangshun) [inline website/package.json]: @sveinpg 

Change this line to:

```
"docusaurus": "git://github.com/facebook/docusaurus.git#127500532103ee9096de9f202866546969ea107d"
```

Refer to https://stackoverflow.com/a/18049685/1751946 for more info.

**Comment by yangshun** [comment]
Docusaurus v1.5.0 has been released - https://github.com/facebook/Docusaurus/blob/master/CHANGELOG.md#150---2018-10-13

@sveinpg was trying out the trunk version yesterday and he showed me that the subcategories worked ok!

unknown-1

Thanks for your patience!

cc @markerikson @timdorr

**Comment by markerikson** [comment]
Okay, I just took a look at the current contents.  It looks great!

I think this is more than sufficient to merge into the `docs-rebuild` branch, and we can open up some more issues to figure out what else actually needs to be done.

@sveinpg , thank you VERY much for some excellent work!

**Pull request merged**